c/src/include/libsbp/internal/unpacked/common.h

```diff
@@ -0,0 +1,298 @@
+  if (!SBP_CAN_UNPACK(ctx, u64)) { return false; }
+  *v = 0;
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 0);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 8);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 16);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 24);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 32);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 40);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 48);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 56);
+  return true;
+}
+
+static inline size_t sbp_packed_size_s8(const s8 *v) {
+  (void)v;
+  return 1;
+}
+
+static inline size_t pack_s8(sbp_pack_ctx_t *ctx, const s8 *v) {
+  return pack_u8(ctx, (const u8 *)v);
+}
+
+static inline size_t unpack_s8(sbp_unpack_ctx_t *ctx, s8 *v) {
+  return unpack_u8(ctx, (u8 *)v);
+}
+
+static inline size_t sbp_packed_size_s16(const s16 *v) {
+  (void)v;
+  return 2;
+}
+
+static inline size_t pack_s16(sbp_pack_ctx_t *ctx, const s16 *v) {
+  return pack_u16(ctx, (const u16 *)v);
+}
+
+static inline size_t unpack_s16(sbp_unpack_ctx_t *ctx, s16 *v) {
+  return unpack_u16(ctx, (u16 *)v);
+}
+
+static inline size_t sbp_packed_size_s32(const s32 *v) {
+  (void)v;
+  return 4;
+}
+
+static inline size_t pack_s32(sbp_pack_ctx_t *ctx, const s32 *v) {
+  return pack_u32(ctx, (const u32 *)v);
+}
+
+static inline size_t unpack_s32(sbp_unpack_ctx_t *ctx, s32 *v) {
+  return unpack_u32(ctx, (u32 *)v);
+}
+
+static inline size_t sbp_packed_size_s64(const s64 *v) {
+  (void)v;
+  return 8;
+}
+
+static inline size_t pack_s64(sbp_pack_ctx_t *ctx, const s64 *v) {
+  return pack_u64(ctx, (const u64 *)v);
+}
+
+static inline size_t unpack_s64(sbp_unpack_ctx_t *ctx, s64 *v) {
+  return unpack_u64(ctx, (u64 *)v);
+}
+
+static inline size_t sbp_packed_size_char(const char *v) {
+  (void)v;
+  return 1;
+}
+
+static inline size_t pack_char(sbp_pack_ctx_t *ctx, const char *v) {
+  return pack_u8(ctx, (const uint8_t *)v);
+}
+
+static inline size_t unpack_char(sbp_unpack_ctx_t *ctx, char *v) {
+  return unpack_u8(ctx, (uint8_t *)v);
+}
+
+static inline size_t sbp_packed_size_float(const float *v) {
+  (void)v;
+  return 4;
+}
+
+static inline size_t pack_float(sbp_pack_ctx_t *ctx, const float *v) {
+  if (!SBP_CAN_PACK(ctx, float)) { return false; }
+  memcpy(&ctx->buf[ctx->offset], v, 4);
+  ctx->offset += 4;
+  return true;
+}
+
+static inline size_t unpack_float(sbp_unpack_ctx_t *ctx,
+                                      float *v) {
+  if (!SBP_CAN_UNPACK(ctx, float)) { return false; }
+  memcpy(v, &ctx->buf[ctx->offset], 4);
+  ctx->offset += 4;
+  return true;
+}
+
+static inline size_t sbp_packed_size_double(const double *v) {
+  (void)v;
+  return 8;
+}
+
+static inline size_t pack_double(sbp_pack_ctx_t *ctx,
+                                     const double *v) {
+  if (!SBP_CAN_PACK(ctx, double)) { return false; }
+  memcpy(&ctx->buf[ctx->offset], v, 8);
```

Comment:
Double's are represented the same on big/little endian?

Reply:
:+1: yeah, no, that needs to be done as well